src/pocketmine/entity/Villager.php

```diff
@@ -73,4 +108,40 @@ public function getProfession() : int{
 	public function isBaby() : bool{
 		return $this->getGenericFlag(self::DATA_FLAG_BABY);
 	}
+
+	public function setCanTrade(bool $value = true) : void{
+		$this->canTrade = $value;
+	}
+
+	public function isCanTrade() : bool{
+		return $this->canTrade;
+	}
+
+	public function setPlayerEntityRuntimeId(int $entityRuntimeId) : void{
+		$this->propertyManager->setLong(self::DATA_TRADING_PLAYER_EID, $entityRuntimeId);
+	}
+
+	public function removePlayerEntityRuntimeId() : void{
+		$this->propertyManager->removeProperty(self::DATA_TRADING_PLAYER_EID);
```

Comment:
`removeProperty()` should be avoided because there's no way to instruct the client to remove a property. Set this to `0` instead.